app/src/main/java/com/vero/tasky/agenda/data/alarmhandler/AlarmHandlerImpl.kt

```diff
@@ -36,16 +36,18 @@ class AlarmHandlerImpl(
             putExtra(AlarmHandler.TITLE, data.title)
             putExtra(AlarmHandler.TYPE, data.type)
         }
-        alarmManager.setExactAndAllowWhileIdle(
-            AlarmManager.RTC_WAKEUP,
-            LocalDateTimeConverter.localDateTimeToLong(data.time) * 1000,
-            PendingIntent.getBroadcast(
-                context,
-                data.itemId.hashCode(),
-                intent,
-                PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE
+        if (data.time.isAfter(LocalDateTime.now())) {
```

Comment:
What happens here, if the user saves the data in timezone A and then this code is run in timezone B? Whenever you persistently save time values or sync these with a backend, you need to take care of also adjusting the timezones. Typically, you save all times in UTC time, but would then need to convert these to the local user's timezone when displaying them